pkg/file/joiner/internal/job.go

```diff
@@ -110,12 +107,24 @@ func (j *SimpleJoinerJob) nextReference(level int) error {
 	chunkAddress := swarm.NewAddress(data[cursor : cursor+swarm.SectionSize])
 	err := j.nextChunk(level-1, chunkAddress)
 	if err != nil {
-		return err
+		// if the last write is a "dangling chunk" the data chunk will have been moved
+		// to an intermediate level. In this edge case, the error must be suppressed,
+		// and the cursor manually to data length boundary to terminate the loop in
+		// the calling frame.
+		if j.readCount+int64(len(data)) == j.spanLength {
+			j.cursors[level] = len(j.data[level])
+			//j.dataC <- data
+			err = j.sendChunkToReader(data)
+			return err
+		}
+		if err != io.EOF {
+			j.logger.Debugf("error in chunk join for %v: %v", chunkAddress, err)
+		}
 	}
 
 	// move the cursor to the next reference
 	j.cursors[level] += swarm.SectionSize
-	return nil
+	return err
```

Comment:
there should not be an error if the execution gets here. happy path no?

Reply:
not quite;

We want the error to reach the end of the go-routine in `NewSimpleJoinerJob`, so that `Read` calls after EOF actually can return an `EOF` error.

Reply:
... which I hereby did.

Reply:
Not sure why this was marked as resolved. I see now that the logic of this function has changed extensively...